### PR TITLE
Docs and examples update for new API

### PR DESCRIPTION
**Pull request by texodus**
* Updates the `docs` project to build against the new API
* Updates the `examples` projects as well, removing deprecated examples.
* Updates the `perspective-python` API for feedback from @timkpaine
* Adds Windows and OSX build support to CI

### CODE REVIEW

File: rust/perspective-python/perspective/handlers/starlette.py [hunk 1]
```diff
@@ -11,52 +11,42 @@
...
-        ...     handler = PerspectiveStarletteHandler(manager, websocket)
+        ...     handler = PerspectiveStarletteHandler(server, websocket)
         ...     await handler.run()
         ... app.add_api_websocket_route('/websocket', endpoint)
     """
 
     def __init__(self, **kwargs):
+        self._server = kwargs.pop("perspective_server")
+        self._websocket = kwargs.pop("websocket")
         super().__init__(**kwargs)
```
Comment (timkpaine) [inline rust/perspective-python/perspective/handlers/starlette.py:37]: not needed, also don't need to use kwargs (no inheritance anymore)

File: rust/perspective-python/perspective/handlers/aiohttp.py [hunk 2]
```diff
@@ -25,41 +24,31 @@ class PerspectiveAIOHTTPHandler(PerspectiveHandlerBase):
...
         ...    await handler.run()
 
         >>> app = web.Application()
         >>> app.router.add_get("/websocket", websocket_handler)
     """
 
     def __init__(self, **kwargs):
+        self.server = kwargs.pop("perspective_server")
+        self._request = kwargs.pop("request")
         super().__init__(**kwargs)
```
Comment (timkpaine) [inline rust/perspective-python/perspective/handlers/aiohttp.py:39]: not needed, also don't need to use kwargs (no inheritance anymore)

File: rust/perspective-python/perspective/handlers/tornado.py [hunk 3]
```diff
@@ -29,35 +26,38 @@ class PerspectiveTornadoHandler(PerspectiveHandlerBase, WebSocketHandler):
...
         ...     (r"/", MainHandler),
         ...     (r"/websocket", PerspectiveTornadoHandler, {
-        ...         "manager": MANAGER,
+        ...         "perspective_server": server,
         ...         "check_origin": True
         ...     })
         ... ])
     """
 
     def __init__(self, *args, **kwargs):
```
Comment (timkpaine) [inline rust/perspective-python/perspective/handlers/tornado.py:41]: no init needed, args passed in via `initialize` for tornado: https://github.com/finos/perspective/pull/2664/files#diff-f6dab3bf154b1234938c04af999f4d052c4d9f6e5fa7220cb789ccf3512e15acR19

File: rust/perspective-python/perspective/core/exception.py [hunk 4]
```diff
@@ -1,17 +0,0 @@
...
-#  ┃ █      ██████ █  ▀█▄       █ ██████      █      ███▌▐███ ███████▄ █       ┃
-#  ┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┫
-#  ┃ Copyright (c) 2017, the Perspective Authors.                              ┃
-#  ┃ ╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌ ┃
-#  ┃ This file is part of the Perspective library, distributed under the terms ┃
-#  ┃ of the [Apache License 2.0](https://www.apache.org/licenses/LICENSE-2.0). ┃
-#  ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
-
-
-class PerspectiveError(Exception):
```
Comment (timkpaine) [inline rust/perspective-python/perspective/core/exception.py:14]: I think https://github.com/finos/perspective/pull/2663 is better overall, allows us to differentiate engine errors from non-engine errors while still catching both

File: rust/perspective-python/perspective/handlers/__init__.py [hunk 5]
```diff
@@ -10,19 +10,24 @@
 #  ┃ of the [Apache License 2.0](https://www.apache.org/licenses/LICENSE-2.0). ┃
 #  ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
 
-from .common import PerspectiveHandlerBase
+__all__ = [
```
[the remark this answers is not in the document]
Comment (texodus) [reply rust/perspective-python/perspective/handlers/__init__.py]: Resolved. This was a `ruff` warning, which is not resolved by dynamically assembling `__all__`. We have dozens of others though and this is not the right PR for addressing warnings anyway.

File: rust/perspective-python/perspective/core/exception.py (hunk 4 above)
Comment (texodus) [reply rust/perspective-python/perspective/core/exception.py:14]: You can still differentiate "engine" errors (e.g. `PerspectiveError`) from everything else (e.g. not `PerspectiveError`).

If we did our job correctly, there is no more Python code to throw non-"engine" errors. As of now, the only explicit errors left in Python are validation errors.

**Pull request merged**